iot_relay_server/app/ai_agent.py

```
@@ -65,226 +75,280 @@ def _apply_rules(sensor_data: dict, weather: dict) -> list[dict]:
-        decisions.append(_make_decision("shading", control_state["shading"], f"야간 외부 {ext_temp}C — 동해 방지 보온커튼 100%.", "emergency", "rule"))
+        decisions.append(_record_decision("shading", control_state["shading"], f"야간 외부 {ext_temp}C — 동해 방지 보온커튼 100%.", "emergency", "rule"))
     elif is_night and ext_temp is not None and ext_temp < 10:
         ins = max(control_state["shading"]["insulation_pct"], 70)
         control_state["shading"]["insulation_pct"] = ins
-        decisions.append(_make_decision("shading", control_state["shading"], f"야간 외부 {ext_temp}C — 보온커튼 {ins}%.", "medium", "rule"))
+        decisions.append(_record_decision("shading", control_state["shading"], f"야간 외부 {ext_temp}C — 보온커튼 {ins}%.", "medium", "rule"))
+
+    return decisions
+
+
+def _apply_general_rules(sensor_data: dict, weather: dict) -> list[dict]:
```

Ruff (0.15.9)

[warning] Unused function argument: `weather`

(ARG001)